lib/Sema/SemaExpr.cpp

```diff
@@ -1731,9 +1731,13 @@ Sema::ActOnStringLiteral(ArrayRef<Token> StringToks, Scope *UDLScope) {
   // Get an array type for the string, according to C99 6.4.5.  This includes
   // the nul terminator character as well as the string length for pascal
   // strings.
+  CheckedArrayKind ArrayKind = getCurScope()->isCheckedScope() ?
+    CheckedArrayKind::Checked : CheckedArrayKind::Unchecked;
+
   QualType StrTy = Context.getConstantArrayType(CharTyConst,
-                                 llvm::APInt(32, Literal.GetNumStringChars()+1),
-                                 ArrayType::Normal, 0);
+                                                llvm::APInt(32, Literal.GetNumStringChars() + 1),
```

Comment:
Is this constant `32` correct? Surely it should be the same as the width of `size_t` which you can get from… somewhere?